### ISSUE TITLE
[Feature] Collapse trees

### ISSUE THREAD

**Issue by strigona-worksight** [traxium/tabtree#4]
Like "Tree Style Tabs", it'd be nice to be able to collapse trees (as can be seen in this screenshot: http://cdn3.howtogeek.com/wp-content/uploads/2015/01/img_54c042e59013c.png ).

**Comment by moszeed** [comment]
+1

**Comment by processor286** [comment]
Yes, the tab column quickly ends up being deeper than the screen, necessitating scrolling the tab list. If trees could collapse, it would be easier to navigate.

**Comment by miscfiles** [comment]
I agree. I'm finding tabtree much better than TST so far, but collapsible trees is an important omisssion.

**Comment by CattiestPenny** [comment]
+1

**Comment by ender4** [comment]
Agreed. Collapsing trees is a core feature of Tree Style Tab. If you are trying to replace it, then collapsing trees is necessary. Similarly the context shortcuts for closing a tree, closing children, closing other trees, etc. are also necessary IMO (each toggleable in settings).

**Comment by krvkir** [comment]
Agreed. Now it's only thing which stops me to switch from Tree Style Tab to this extension.

If search would work through collapsed tabs (and there will be hotkey to focus on search input box) it would be even better.

**Comment by traxium** [comment]
@krvkir, Ctrl+Shift+Alt+F focuses the tab search input box (or brings it up when it's hidden).

**Comment by Cosolom** [comment]
+1. @traxium, Do you plannig this feature?

**Comment by traxium** [comment]
@Cosolom, yes. But it can't be done soon.

**Comment by timss** [comment]
Second only to vertical tabs and the tree feature itself, this is probably the most sought after feature when people want something like Tree Style Tab. Same would be true for Tab Tree. 

Will probably try to switch when it's added. Thanks for the work so far.

**Comment by haarp** [comment]
+1

**Comment by eternaleye** [comment]
:+1:

**Comment by tomdavey** [comment]
+1

**Comment by Gandalf99** [comment]
Nice extension, traxium. Thank you. 
I agree with these comments. Collapsing trees is important. I may go back to TST until this is implemented. You wrote last September that you wanted to implement it. Can you give an estimate of when? A week? A month? A year? . . .

**Comment by meekreau** [comment]
+1

**Comment by padthaitofuhot** [comment]
+1 
Great extension @traxium, I use it every day.  Any ETA on collapsible trees?  Anything the community can do to help motivate (airdrop beers? gummybears? paypal pings?)

**Comment by traxium** [comment]
I already explained somewhere why I'm not doing collapsible trees right now but I'll say it again with additional details. Firstly, I want to do collapsible trees very very much. I created a branch in git called "collapsible" a long time ago and even committed something there. But something bad happened to me back then (let's say an injury). I don't think you want to know details. When I almost recovered from it something bad happened to me again (let's say a chill) and I even was in hospital for two weeks.

What stops me now?
1. Adding collapsible trees affects almost every aspect of the add-on: moving, closing, opening, selecting tabs. So adding collapsible trees can't be done in a day or two. But as we all know there is an important bug — with Intel graphics and Hardware acceleration (at this point I want to try to use the native tabs instead of extremely buggy `<tree>` element that is guilty of this bug and I already have something done in that direction). And there is an important issue with Tab Groups. Although they will be deprecated they are going to continue to live as an add-on. And many people as it seems want to continue to use them. The bug and the issue with Tab Groups require a lot of coding in order to be fixed because they also affect many aspects of the add-on. So it's not rational to add collapsible trees and to resolve the other aforementioned problems separately. It would be wiser to resolve them all together. And It's not something I can do when I have a break from other activities.
2. When I have enough money in order not to work for a couple of months I'll do what my soul wants regarding Tab Tree, I'll complete it.
3. [The Future of Developing Firefox Add-ons](https://blog.mozilla.org/addons/2015/08/21/the-future-of-developing-firefox-add-ons/) doesn't encourage me at all to contribute to something that's going to be deprecated. It's just not possible to do all this using silly High-Level or Low-Level APIs or even worse WebExtensions API. But I hope I'll do what I want before everything is deprecated.

**Comment by timss** [comment]
Thank you for your very personal response, @traxium. Do take the time you need. Wishing you the best of health!

3. We're facing the same problem with Vimperator. Mozilla did hint that they'll add more functionality to support complex addons, but I'm not holding my breath.

**Comment by dotbugfix** [comment]
+1

**Comment by Nomen-Heroum** [comment]
+1
